File: backend/src/main/java/com/knhlje/cafe/model/dto/Favorite.java
package com.ssafy.cafe.model.dto;

import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Favorite {

	private Integer id;
	private String userId;
	private Integer productId;
	
	@Builder
	public Favorite(Integer id, String userId, Integer productId) {
		super();
		this.id = id;
		this.userId = userId;
		this.productId = productId;
	}

	public Favorite(String userId, Integer productId) {
		this.userId = userId;
		this.productId = productId;
	}
}

File: android/SmartStoreUI/app/src/main/java/com/knhlje/smartstore/dto/Favorite.kt
package com.knhlje.smartstore.dto

import java.io.Serializable

data class Favorite(
    val productId: Int,
    val userId: String
): Serializable{
    var id: Int = 0

    constructor(id: Int, productId: Int, userId: String): this(productId, userId){
        this.id = id
    }
}